src/controller/CHIPDevice.h

```diff
@@ -411,6 +423,31 @@ class DLL_EXPORT Device : public Messaging::ExchangeDelegate, public SessionEsta
     CHIP_ERROR EstablishConnectivity(Callback::Callback<OnDeviceConnected> * onConnection,
                                      Callback::Callback<OnDeviceConnectionFailure> * onFailure);
 
+    /*
+     * This function can be called to register callbacks for when a secure session is ready.
+     *
+     * If the device doesn't have operational credentials, and is under commissioning process,
+     * PASE keys will be used for secure session.
+     *
+     * If the device has been commissioned and has operational credentials, callbacks will be queued up.
+     *
+     * On establishing the session, the callback function `onConnection` will be called. If the
+     * session setup fails, `onFailure` will be called.
+     *
+     * If the session already exists, `onConnection` will be called immediately.
+     */
+    void EnqueueConnectionHandlersIfNeeded(Callback::Callback<OnDeviceConnected> * onConnection,
```

Comment:
Could we avoid methods with 'if' in the name? It is prefferable to have `DoFoo()` instead of `DoFooIfBar`. In this case "IfNeeded" seems a bit vague ... when is it needed? Why can't the calling code do `If (Needed()) { EnqueueConnectionHandlers(); }`